portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,50 +14,48 @@
 
 package com.google.sps.servlets;
 
-import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Arrays;
 import java.sql.Timestamp;
 import com.google.gson.Gson;
-import com.google.appengine.api.datastore.PreparedQuery;
-import com.google.appengine.api.datastore.Query;
-import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory; 
-import com.google.appengine.api.datastore.FetchOptions;
-
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
-        Query query = new Query("Task");
-
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-        PreparedQuery results = datastore.prepare(query);
+        if (request.getParameter("maxcomments") == null)
+        response.sendError(response.SC_BAD_REQUEST, "maxComment parameter missing");
 
-        List<Entity> newResults = new ArrayList<Entity>();
-        newResults = results.asList(FetchOptions.Builder.withLimit(5));
+        int maxcomments = Integer.parseInt(request.getParameter("maxcomments"));
 
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(maxcomments));
```

Comment:
If possible, please use ImmutableList instead. We don't expect changes on this list right?

Reply:
Yeh, its better if it can't be changed.
Done!